src/webapi/html_elements/select.rs

```diff
@@ -0,0 +1,181 @@
+///
+/// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/select)
+// https://html.spec.whatwg.org/#htmlselectelement
+#[derive(Clone, Debug, PartialEq, Eq, ReferenceType)]
+#[reference(instance_of = "HTMLSelectElement")]
+#[reference(subclass_of(EventTarget, Node, Element, HtmlElement))]
+pub struct SelectElement(Reference);
+
+impl IEventTarget for SelectElement {}
+impl INode for SelectElement {}
+impl IElement for SelectElement {}
+impl IHtmlElement for SelectElement {}
+
+impl SelectElement {
+    /// Returns the `Some(index)` of the first selected item, if any, or `None` if there is no selected item.
+    // https://html.spec.whatwg.org/#the-select-element:dom-select-selectedindex
+    pub fn selected_index(&self) -> Option<i32> {
+        let si = js! (
+            return @{self}.selectedIndex;
+        ).try_into().unwrap();
+        if si < 0 {
+            None
+        } else {
+            Some(si)
+        }
+    }
+
+    /// Change selected index to the given value.
+    // https://html.spec.whatwg.org/#the-select-element:dom-select-selectedindex
+    pub fn set_selected_index(&self, selected_index: Option<i32>) {
+        let selected_index = selected_index.unwrap_or(-1);
+        js!{
+            @(no_return)
+            @{self}.selectedIndex = @{selected_index};
+        }
+    }
+
+    /// Returns the `Some(value)` of the first selected item, if any, or `None` if there is no selected item.
+    // https://html.spec.whatwg.org/#the-select-element:dom-select-value
+    pub fn value(&self) -> Option<String> {
+        js!(
+            var self = @{self};
+            if (self.selectedIndex < 0) {
+                return null;
+            } else {
+                return self.value;
+            }
+        ).try_into().unwrap()
+    }
+
+    /// Change the selected value to the given value.
+    // https://html.spec.whatwg.org/#the-select-element:dom-select-value
+    pub fn set_value(&self, value: Option<String>) {
+        match value{
+            Some(value) => js!{
+                @(no_return)
+                @{self}.value = @{value};
+            },
+            None => js!{
+                @(no_return)
+                @{self}.selectedIndex = -1;
+            }
+        };
+    }
+
+    /// Indicates whether multiple items can be selected
+    // https://html.spec.whatwg.org/#the-select-element:dom-select-multiple
+    pub fn multiple(&self) -> bool {
+        js!(
+            return @{self}.multiple;
+        ).try_into().unwrap()
+    }
+
+    /// An [HtmlCollection](struct.HtmlCollection.html) representing
+    /// the set of <option> elements that are selected.
+    // https://html.spec.whatwg.org/#the-select-element:dom-select-selectedoptions
+    pub fn selected_options(&self) -> HtmlCollection {
+        js!(
+            return @{self}.selectedOptions;
+        ).try_into().unwrap()
+    }
+
+    /// A convenient method to get values of all selected <option> elements
```

Comment:
`convenient` -> didn't you mean `convenience`? (:

Also `<option>` here is unquoted too, so it will break the docs. (Ditto for the next method.)

Reply:
* What I means is `convenient method` (`adj+noun`). But my English skill is just average, so I am not sure any thing that involves `grammar`.
* About: `<option>`: Do you have a method to detect things like this? After this catch by you, I actually ran a `cargo doc`, go seeing the result docs, and came back to correct the source code docs... then repeat the tasks until not seeing something weird in the result docs.

Reply:
1. In these kinds of contexts you'd usually use `convenience` instead of `convenient` (:

2. I don't; I just usually catch these kinds of issues during the review. Usually you can just simply run `cargo doc`, open the docs and look through the docs for the code you've added/changed if it looks correct.